tools/odbc/driver.cpp

```diff
@@ -18,34 +18,34 @@ using std::string;
 
 SQLRETURN duckdb::FreeHandle(SQLSMALLINT handle_type, SQLHANDLE handle) {
 	if (!handle) {
-		return SQL_ERROR;
+		return SQL_INVALID_HANDLE;
 	}
 
 	switch (handle_type) {
 	case SQL_HANDLE_DBC: {
-		auto *hdl = (duckdb::OdbcHandleDbc *)handle;
+		auto *hdl = static_cast<duckdb::OdbcHandleDbc *>(handle);
 		delete hdl;
 		return SQL_SUCCESS;
 	}
 	case SQL_HANDLE_DESC: {
-		auto *hdl = (duckdb::OdbcHandleDesc *)handle;
+		auto *hdl = static_cast<duckdb::OdbcHandleDesc *>(handle);
 		hdl->dbc->ResetStmtDescriptors(hdl);
 		delete hdl;
-		return SQL_ERROR;
+		return SQL_SUCCESS;
```

Comment:
Oh, just curious: why is this not a `SQL_ERROR`? Was this a bug?

Reply:
Yes I think this was a bug, this function should only return `SQL_ERROR` if [the handle is still valid](https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlfreehandle-function?view=sql-server-ver16#returns), and in this case it should have successfully been deleted.